src/main/webapp/script.js

```diff
@@ -11,3 +11,43 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+
+function getEvent() {
+  const urlParams = new URLSearchParams(window.location.search);
+  const event_id = urlParams.get('event_id');
+  fetch('/event?event_id=' + event_id).then(response => response.json()).then();
+}
+
+function getUser() {
+  const urlParams = new URLSearchParams(window.location.search);
+  const user_id = urlParams.get('user_id');
+  fetch('/user?user_id=' + user_id).then(response => response.json()).then();
+}
+
+/* Add all fetched events to the page. */
+function getEvents() {
+  fetch('/events').then(response => response.json()).then(events => events.forEach(addEvent));
+}
+
+/* Add all fetched events which match the search to the page. */
+function getSearchEvents() {
+  fetch('/events?' + new URLSearchParams({
+    search: search,
+  })).then(response => response.json()).then(events => events.forEach(...));
+}
+
+function getGCalendarEvents() {
+  fetch('/events/gcalendar').then(response => response.json()).then(events => events.forEach(...));
```

Comment:
This one should be done differently, to cope with the token thing. But it is okay, I will implement that.